src/java.base/share/classes/java/text/CompactNumberFormat.java

```diff
@@ -217,6 +217,8 @@ public final class CompactNumberFormat extends NumberFormat {
     @java.io.Serial
     private static final long serialVersionUID = 7128367218649234678L;
 
+    // Non-transient / non-static fields should be added to hashCode impl
+
```

Comment:
Did you mean `equals()`? `hashCode()` may not have all the non-transient fields.